### PR TITLE
Add readme

### PR DESCRIPTION
**Pull request by xiaoling-yi**
This RP adds a detailed readme for the GEMM accelerator, including:
* Microarchitecture
* Functional description
* Unit test description

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,69 @@
 # Integer GEMM Accelerator for SNAX
-This repository contains the RTL to build an 8-bit integer GEMM accelerator 
-to integrate into the [SNAX core](https://github.com/KULeuven-micas/snitch_cluster). 
+This repository contains the RTL to build an 8-bit integer GEMM (General Matrix Multiply) accelerator 
+to integrate into the [SNAX core](https://github.com/KULeuven-micas/snitch_cluster). This repository contains the chisel src file and unit test of the GEMM accelerator.
```
Comment (JosseVanDelm) [inline README.md:3]:
```suggestion
to integrate into the [SNAX core](https://github.com/KULeuven-micas/snitch_cluster). This repository contains the chisel sources and unit tests for the GEMM accelerator.
```

File: README.md [hunk 2]
```diff
@@ -1,19 +1,69 @@
...
+![](./docs/datalayout.png)
+
+The Base GEMM function definition pseudocode is shown below.
+```
+bool gemm(signed byte *A,signed byte *B, int * C)
+```
+
+### Large GEMM
+The Large GEMM is built on the Base GEMM. It takes in the M, K, and N configurations.
+In this case, the size of matrix A is (M * meshRow, K * tileSize) and the size of matrix B is (K * tileSize, N * meshCol). The size of result matrix C is (M * meshRow, N * meshCol). The GEMM accelerator uses Block matrix multiplication [](https://en.wikipedia.org/wiki/Block_matrix#Block_matrix_multiplication) method to implement matrix multiplication in which the matrix sizes are much larger than the physical GEMM array. To get the right results, matrixes should have the right layout in memory. Below is an example data layout in memory for M = 2, K = 2, and N = 2.
```
Comment (JosseVanDelm) [inline README.md:37]:
```suggestion
In this case, the size of matrix A is (M * meshRow, K * tileSize) and the size of matrix B is (K * tileSize, N * meshCol). The size of result matrix C is (M * meshRow, N * meshCol). The GEMM accelerator uses [Block matrix multiplication](https://en.wikipedia.org/wiki/Block_matrix#Block_matrix_multiplication) method to implement matrix multiplication in which the matrix sizes are much larger than the physical GEMM array. To get the right results, matrixes should have the right layout in memory. Below is an example data layout in memory for M = 2, K = 2, and N = 2. 
```

Comment (JosseVanDelm) [reply README.md:37]: Can you split this into multiple lines please?

File: README.md [hunk 3]
```diff
@@ -1,19 +1,69 @@
...
+### Batch Stride Large GEMM
+The Batch Stride Large GEMM is built on the Batch Large GEMM. It takes in three extra strides configuration, eg., strideA, strideB, and strideC, for computing the start matrix address for each batch. 
+
+The Batch Stride Large GEMM function definition pseudocode is shown below.
+```
+bool batchstridelargegemm(int B, int M, int K, int N, int strideA, int strideB, int strideC, signed byte *A,signed byte *B, int * C)
+```
+
+## Unit test
+This repository also contains some unit tests for each version of GEMM to do the numerical test. The unit tests are also written in Chisel. Firstly, the random input matrixes and random size configurations are generated. Then these matrixes and configurations are input into the GEMM accelerator. After the computation, the output result of the GEMM accelerator will be compared with the golden model.
```
Comment (JosseVanDelm) [inline README.md:63]:
```suggestion
This repository also contains some unit tests for each version of GEMM to do the numerical test. The unit tests are also written in Chisel. Firstly, random input matrices and random size configurations are generated. Then these matrices and configurations are fed into the GEMM accelerator. After the computation, the output result of the GEMM accelerator will be compared with the golden model in Scala.
```

File: README.md [hunk 4]
```diff
@@ -1,19 +1,69 @@
...
+
+## Quick Start
+Following this quick start to set up the Chsiel environment and run the unit test of the GEMM accelerator.
```
Comment (JosseVanDelm) [inline README.md:66]:
```suggestion
## Quickstart
Follow this quickstart to set up the Chisel environment and run the unit tests of the GEMM accelerator.
```

File: README.md [hunk 5]
```diff
@@ -1,19 +1,69 @@
...
+bool batchlargegemm(int B, int M, int K, int N, signed byte *A,signed byte *B, int * C)
+```
+
+### Batch Stride Large GEMM
+The Batch Stride Large GEMM is built on the Batch Large GEMM. It takes in three extra strides configuration, eg., strideA, strideB, and strideC, for computing the start matrix address for each batch. 
+
+The Batch Stride Large GEMM function definition pseudocode is shown below.
+```
+bool batchstridelargegemm(int B, int M, int K, int N, int strideA, int strideB, int strideC, signed byte *A,signed byte *B, int * C)
+```
```
Comment (JosseVanDelm) [inline README.md:60]: Can we add this when this functionality is added maybe?

**Review by JosseVanDelm — CHANGES_REQUESTED** [review]
This is really great!

Please:
* Add figures
* Remove unimplemented stuff (for now!)

File: README.md (hunk 5 above)
Comment (xiaoling-yi) [reply README.md:60]: Sure!

**Comment by JosseVanDelm** [comment]
Closing this for now in favor of #8 and future PRs

**Pull request closed without merging**